### PR TITLE
Add test framework

### PR DESCRIPTION
**Pull request by dansimau**
This adds some basic tests to the ringpop-admin command output.

### CODE REVIEW

File: tests/partitions.t [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+With bootstrap file:
+
+  $  ringpop-admin partitions $TESTDIR/hosts.json
+   Checksum     # Nodes   Sample Host    
+   *   5         127.0.0.1:3000  (glob)
+
+
+Unable to connect to host:
+
+  $  ringpop-admin partitions 127.0.0.1:2999
```
Comment (jwolski) [inline tests/partitions.t]: The fields in this error are provided by TChannel. Is this test going to fail if TChannel changes what they log?

Comment (dansimau) [inline tests/partitions.t]: Yes. We could glob all the lines I guess. Or improve the error output to be more human-readable. Also, note that it's very easy to update the test if the output changes (just do `./run-tests --update` and you can accept any changes you know are valid output).

Comment (jwolski) [inline tests/partitions.t]: Yes, I caught the explanation in the readme after I made this comment.

**Comment by jwolski** [comment]
lgtm. merge at will

**Comment by jwolski** [comment]
Also, how about a pre-commit hook that runs the suite?

**Comment by dansimau** [comment]
@jwolski done, added pre-commit hook. Merging.

**Comment by jwolski** [comment]
@dansimau  u r a boss

File: tests/partitions.t (hunk 1 above)
Comment (CorgiMan) [inline tests/partitions.t]: My tests failed because of this so I think the test is a bit too strict like this

Comment (dansimau) [inline tests/partitions.t]: What was the diff?

File: tests/partitions.t [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  Error while fetching node stats: { [TchannelSocketError: tchannel socket error (ECONNREFUSED from connect): connect ECONNREFUSED]
+    type: 'tchannel.socket',
+    message: 'tchannel socket error (ECONNREFUSED from connect): connect ECONNREFUSED',
+    hostPort: null,
+    direction: 'out',
+    remoteAddr: null,
+    name: 'TchannelSocketError',
+    socketRemoteAddr: '127.0.0.1:2999',
```
Comment (CorgiMan) [inline tests/partitions.t]: should this port number be globbed?

**Pull request merged**